lib/r10k/environment/with_modules.rb

```diff
@@ -49,15 +49,34 @@ def modules
     return @modules if puppetfile.nil?
 
     puppetfile.load unless puppetfile.loaded?
+    @modules + puppetfile.modules
+  end
 
-    env_mod_names = @modules.map(&:name)
-    @modules + puppetfile.modules.select { |mod| !env_mod_names.include?(mod.name) }
+  def module_conflicts?(mod_b)
+    conflict = @modules.any? { |mod_a| mod_a.name == mod_b.name }
+    return false unless conflict
+
+    msg_vars = {src: mod_b.origin, name: mod_b.name}
+    msg_error = _('Environment and %{src} both define the "%{name}" module' % msg_vars)
+    msg_continue = _("#{msg_error}. The %{src} definition will be ignored" % msg_vars)
+
+    case conf = @options[:module_conflicts]
```

Comment:
(nit) I liked calling this `conflict_opt`

Reply:
I'll change it back 🙂